tracer/test/Datadog.Trace.Tests/Configuration/ExporterSettingsTests.cs

```diff
@@ -16,66 +17,191 @@ namespace Datadog.Trace.Tests.Configuration
     public class ExporterSettingsTests
     {
         [Fact]
-        public void NoSocketFiles_NoExplicitConfiguration_DefaultsMatchExpectation()
+        public void DefaultValues()
         {
-            var config = Setup(FileExistsMock());
-            Assert.Equal(expected: TracesTransportType.Default, actual: config.TracesTransport);
-            Assert.Equal(expected: MetricsTransportType.UDP, actual: config.MetricsTransport);
-            Assert.Equal(expected: new Uri($"http://127.0.0.1:8126"), actual: config.AgentUri);
-            Assert.Equal(expected: 8125, actual: config.DogStatsdPort);
-            Assert.False(config.PartialFlushEnabled);
-            Assert.Equal(expected: 500, actual: config.PartialFlushMinSpans);
+            var settings = new ExporterSettings();
+            CheckDefaultValues(settings);
+        }
+
+        [Fact]
+        public void AgentUri()
+        {
+            var param = "http://someUrl";
+            var uri = new Uri(param);
+            var settings = new ExporterSettings { AgentUri = uri };
+            var settingsFromSource = Setup("DD_TRACE_AGENT_URL", param);
+
+            AssertHttpIsConfigured(settingsFromSource, uri);
+            AssertHttpIsConfigured(settings, uri);
+        }
+
+        [Fact]
+        public void UnixAgentUriOnWindows()
+        {
+            var param = @"C:\temp\someval";
+            var uri = new Uri($"unix://{param}");
+
+            var settingsFromSource = Setup("DD_TRACE_AGENT_URL", $"unix://{param}");
+            AssertUdsIsConfigured(settingsFromSource, param.Replace("\\", "/"));
+            settingsFromSource.AgentUri.Should().Be(uri);
         }
 
         [Fact]
-        public void PartialFlushVariables_Populated()
+        public void InvalidAgentUrlShouldNotThrow()
```

Comment:
If you set `ExporterSettings.AgentUri` from code, that will end up throwing right? Is there a test case for that?

Reply:
No it won't as the property isn't a string but a Uri. Thus the Uri has already been created. So this part cannot fail